munge_aggregates.js

```diff
@@ -0,0 +1,325 @@
+        }
+
+        if (!resultsTaxonomy[implementation_id][version][full_name]) {
+            resultsTaxonomy[implementation_id][version][full_name] = {
+                slug,
+                full_name,
+                outcome,
+            };
+        }
+    }
+
+    for (const [implementation_id, versions] of Object.entries(resultsTaxonomy)) {
+        outputFrontmatter(`content/results/${implementation_id}/_index.md`, {
+            implementation_id,
+            title: implementation_id
+        });
+
+        for (const [version, tests] of Object.entries(versions)) {
+            outputFrontmatter(`content/results/${implementation_id}/${version}/_index.md`, {
+                implementation_id,
+                version,
+                title: version
+            });
+
+            for (const test of Object.values(tests)) {
+                outputFrontmatter(`content/results/${implementation_id}/${version}/${test.slug}/_index.md`, {
+                    ...test,
+                    implementation_id,
+                    version,
+                    title: test.full_name
+                });
+            }
+        }
+    }
+
+    // Close the database connection when you're done
+    db.close((err) => {
+        if (err) {
+            console.error(err.message);
+        }
+        console.log('Closed the database connection.');
+    });
+}
+
+const slugify = (str) => {
+    // https://byby.dev/js-slugify-string
+    return String(str)
+        .normalize('NFKD') // split accented characters into their base characters and diacritical marks
+        .replace(/[\u0300-\u036f]/g, '') // remove all the accents, which happen to be all in the \u03xx UNICODE block.
+        .trim() // trim leading or trailing whitespace
+        .toLowerCase() // convert to lowercase
+        .replace(/\s+/g, '_') // replace spaces with underscore
+        .replace(/_+/g, '_') // remove consecutive underscores
+        .replace(/[\/]/g, "__")
+        .replace(/[^a-z0-9 -]/g, '-') // remove non-alphanumeric characters
+}
+
+const outputJSON = (p, data) => {
+    const json = JSON.stringify(data, null, 2);
+    const fullPath = `${hugoOutput}/${p}`;
+
+    const folders = path.dirname(fullPath);
+    if (!fs.existsSync(folders)) {
+        fs.mkdirSync(folders, { recursive: true });
+    }
+
+    fs.writeFileSync(fullPath, json);
+}
+
+const outputFrontmatter = (p, data) => {
+    const fullPath = `${hugoOutput}/${p}`;
+
+    // TODO: implement update frontmatter
+
+    const folders = path.dirname(fullPath);
+    if (!fs.existsSync(folders)) {
+        fs.mkdirSync(folders, { recursive: true });
+    }
+
+    // if file exists, load it with gray matter
+    const content = {
+        content: "",
+        data: {}
+    }
+    if (fs.existsSync(fullPath)) {
+        const existing = matter.read(fullPath);
+        content.content = existing.content;
+        content.data = existing.data;
+    }
+    content.data = { ...content.data, ...data };
```

Comment:
So we'll have to garbage collect at some point, but I'd like to preserve the files instead of overwriting:
this way, we can have humans add content and details, since each page (like "kubo", or "test-XYZ") is a regular page.